src/AutoRest.CSharp/LowLevel/Generation/LowLevelRestClientWriter.cs

```diff
@@ -191,19 +192,49 @@ private void WriteStatusCodeSwitch(CodeWriter writer, RestClientMethod clientMet
                             writer.Line($"case int s when s >= {statusCode.Family * 100:L} && s < {statusCode.Family * 100 + 100:L}:");
                         }
                     }
-                    writer.Line($"return {messageVariable}.{nameof(HttpMessage.Response)};");
+                    var returnType = clientMethod.ReturnType;
+                    if (responseBody != null && responseBody is ConstantResponseBody body && returnType != null)
+                    {
+                        writer.Append($"return {typeof(Response)}.FromValue(");
+                        writer.WriteReferenceOrConstant(body.Value);
+                        writer.Append($", {messageVariable}.{nameof(HttpMessage.Response)});");
+                    }
+                    else
+                    {
+                        writer.Line($"return {messageVariable}.{nameof(HttpMessage.Response)};");
+                    }
                 }
 
                 writer.Line($"default:");
-                if (async)
+                if (isNoThrow)
                 {
-                    writer.Line($"throw await {ClientDiagnosticsField}.CreateRequestFailedExceptionAsync({messageVariable}.{nameof(HttpMessage.Response)}).ConfigureAwait(false);");
+                    writer.Append($"return {typeof(ResponseWithError)}.FromError<{clientMethod.ReturnType}>({messageVariable}.{nameof(HttpMessage.Response)}, {GetAwait(async)}{ClientDiagnosticsField}.CreateRequestFailedException{GetAsyncSuffix(async)}({messageVariable}.{nameof(HttpMessage.Response)}){GetConfigureAwait(async)});");
```

Comment:
So the real difference here is `return {typeof(ResponseWithError)}.FromError<{clientMethod.ReturnType}>({messageVariable}.{nameof(HttpMessage.Response)},` vs `throw`, is that correct?

I'd also extract this block into separate method.